src/bin/vip-wp.js

```diff
@@ -72,6 +112,66 @@ commandWrapper( {
 
 		let result;
 		let rl;
+		let subShellRl;
+
+		if ( isSubShell ) {
+			console.log( `Entering WP-CLI subshell mode for ${ formatEnvironment( envName ) } on ${ appName } (${ appId })` );
+
+			subShellRl = readline.createInterface( {
+				input: process.stdin,
+				output: process.stdout,
+				terminal: true,
+				prompt: `${ appName }.${ envName }> `,
+				// TODO make history persistent across sessions for same env
+				historySize: 200,
+			} );
+
+			subShellRl.on( 'line', async line => {
+				console.log( 'received line:', line );
+				subShellRl.pause();
+
+				try {
+					result = await getTokenForCommand( appId, envId, line.replace( 'wp ', '' ) );
+				} catch ( e ) {
+					console.log( e );
+
+					return;
+				}
+
+				const { data: { triggerWPCLICommandOnAppEnvironment: { command: cliCommand, inputToken } } } = result;
+
+				const commandStreams = await launchCommandAndGetStreams( {
+					guid: cliCommand.guid,
+					inputToken: inputToken,
+				} );
+
+				commandStreams.stdoutStream.pipe( process.stdout );
+
+				commandStreams.stdoutStream.on( 'error', err => {
+					// TODO handle this better
+					console.log( err );
+
+					process.exit( 1 );
+				} );
+
+				commandStreams.stdoutStream.on( 'end', () => {
+					subShellRl.resume();
+				} );
+
+				subShellRl.on( 'SIGINT', () => {
+					subShellRl.question( 'Are you sure you want to exit the subshell mode? ', answer => {
```

Comment:
I'm not sure we _need_ a prompt on exit, as there isn't much of a negative to it, and it's easy to get back in.

Eventually we should support catching `SIGINT` and sending that through to the server, to interrupt the command.

Reply:
with this https://github.com/Automattic/vip/pull/280/commits/6f25fab78f6ce5b465e6d1c559d74e1f90dcbe0e it just closes the stream without writing anything :)

Reply:
I was unable to exit the subshell after 6f25fab, so I modified the `SIGINT` handler to close the readline then exit the process in 02d5fc2.